### PR TITLE
koord-descheduler: evict pod by perferred groupverison

### PR DESCRIPTION
**Pull request by LeoLiuYan**
Signed-off-by: LeoLiuYan <929908264@qq.com>

### Ⅰ. Describe what this PR does

Find out the server if support eviction subresource and use preferred groupversion.

### Ⅱ. Does this pull request fix one issue?

NONE

### V. Checklist

- [✅] I have written necessary docs and comments
- [✅] I have added necessary unit tests and integration tests
- [✅] All checks passed in `make test`

### CODE REVIEW

File: pkg/descheduler/controllers/migration/evictor/interpreter.go [hunk 1]
```diff
@@ -37,7 +38,7 @@ var (
 	ErrTooManyEvictions = errors.New("TooManyEvictions")
 )
 
-type FactoryFn func(client kubernetes.Interface) Interface
+type FactoryFn func(client kubernetes.Interface, policyGroupVersion string) Interface
```
Comment (eahydra) [inline pkg/descheduler/controllers/migration/evictor/interpreter.go:41]: It is recommended to keep the parameters of this function type unchanged, and directly modify the implementation-aware eviction policyGroupVersion of NativeEvictor.

**Review by eahydra — CHANGES_REQUESTED** [review]

File: pkg/descheduler/controllers/migration/evictor/interpreter.go (hunk 1 above)
Comment (LeoLiuYan) [reply pkg/descheduler/controllers/migration/evictor/interpreter.go]: Fine. Changed it. 

@eahydra

File: pkg/descheduler/evictions/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+package utils
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+const (
+	EvictionKind           = "Eviction"
+	EvictionGroupName      = "eviction"
```
[the remark this answers is not in the document]
Comment (LeoLiuYan) [reply pkg/descheduler/evictions/utils/utils.go:26]: Yes, I will fix it.  @FillZpp

File: pkg/descheduler/controllers/migration/evictor/evictor_native.go [hunk 3]
```diff
@@ -52,7 +55,18 @@ func (e *NativeEvictor) Evict(ctx context.Context, job *sev1alpha1.PodMigrationJ
 		Name:      pod.Name,
 	}
 	klog.Infof("Try to evict Pod %q with MigrationJob %s", namespacedName, job.Name)
-	err := evictions.EvictPod(ctx, e.client, pod, job.Spec.DeleteOptions)
+	if len(e.policyGroupVersion) == 0 {
+		// refresh policy groupversion
+		policyGroupVersion, err := evictutils.SupportEviction(e.client)
```
Comment (eahydra) [inline pkg/descheduler/controllers/migration/evictor/evictor_native.go:60]: maybe we can call the function `SupportEviction` in `NewNativeEvictor`

**Comment by FillZpp** [comment]
/lgtm

Thanks @LeoLiuYan

**Review by eahydra — APPROVED** [review]
/lgtm
/approve

**Pull request merged**